From ac0037ea5d91beb16ddc530594aba50d6bcf3971 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:29:27 -0400
Subject: [alsa] ice1712: restore AK4xxx volumes on resume

Message-id: <140972823573-254-patchset-jkysela@redhat.com>
Patchwork-id: 90954
O-Subject: [RHEL 7.1 PATCH 253/442] ALSA: ice1712: restore AK4xxx volumes on resume
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Also restore AK4xxx mixer volumes on resume for M-Audio ICE1712-based cards.
This fixes incorrect (sound working) zero mixer volumes after resume.

Signed-off-by: Ondrej Zary <linux@rainbow-software.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 6e61246f5aeaede80e7a00e8b1de7ae07b1315e7)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/ice1712/delta.c b/sound/pci/ice1712/delta.c
index b3eed8d..496dbd0 100644
--- a/sound/pci/ice1712/delta.c
+++ b/sound/pci/ice1712/delta.c
@@ -579,7 +579,8 @@ static struct snd_ak4xxx_private akm_vx442_priv = {
 #ifdef CONFIG_PM_SLEEP
 static int snd_ice1712_delta_resume(struct snd_ice1712 *ice)
 {
- unsigned char akm_backup[AK4XXX_IMAGE_SIZE];
+ unsigned char akm_img_bak[AK4XXX_IMAGE_SIZE];
+ unsigned char akm_vol_bak[AK4XXX_IMAGE_SIZE];
 
  /* init spdif */
  switch (ice->eeprom.subvendor) {
@@ -604,9 +605,11 @@ static int snd_ice1712_delta_resume(struct snd_ice1712 *ice)
 
  /* init codec and restore registers */
  if (ice->akm_codecs) {
-  memcpy(akm_backup, ice->akm->images, sizeof(akm_backup));
+  memcpy(akm_img_bak, ice->akm->images, sizeof(akm_img_bak));
+  memcpy(akm_vol_bak, ice->akm->volumes, sizeof(akm_vol_bak));
   snd_akm4xxx_init(ice->akm);
-  memcpy(ice->akm->images, akm_backup, sizeof(akm_backup));
+  memcpy(ice->akm->images, akm_img_bak, sizeof(akm_img_bak));
+  memcpy(ice->akm->volumes, akm_vol_bak, sizeof(akm_vol_bak));
   snd_akm4xxx_reset(ice->akm, 0);
  }
 
-- 
1.7.1